RecoTracker/LSTCore/src/alpaka/TrackCandidate.h

```diff
@@ -609,6 +641,230 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst {
+      ALPAKA_ASSERT_ACC((alpaka::getWorkDiv<alpaka::Grid, alpaka::Blocks>(acc)[1] == 1) &&
+                        (alpaka::getWorkDiv<alpaka::Grid, alpaka::Blocks>(acc)[2] == 1));
+
+      // 2. Assert that the Block is 2D ([0] dimension must be 1).
+      ALPAKA_ASSERT_ACC((alpaka::getWorkDiv<alpaka::Block, alpaka::Threads>(acc)[0] == 1));
+
+      // Shared memory: Packed best candidate info per logical OT layer (1..11)
+      // Format: [Score (32 bits) | Quintuplet Index (28 bits) | Layer Slot (4 bits)]
+      uint64_t* sharedBestPacked = alpaka::declareSharedVar<uint64_t[lst::kLogicalOTLayers], __COUNTER__>(acc);
+
+      const unsigned int trackCandidateIndex = alpaka::getIdx<alpaka::Grid, alpaka::Blocks>(acc)[0u];
+      const unsigned int nTrackCandidates = candsBase.nTrackCandidates();
+
+      if (trackCandidateIndex >= nTrackCandidates)
+        return;
+
+      // Initialize shared memory once per block
+      if (cms::alpakatools::once_per_block(acc)) {
+        for (int logicalLayerBin = 0; logicalLayerBin < lst::kLogicalOTLayers; ++logicalLayerBin) {
+          sharedBestPacked[logicalLayerBin] = 0;  // 0.0f score, 0 index, 0 slot
+        }
+      }
+      alpaka::syncBlockThreads(acc);
+
+      const LSTObjType trackCandidateType = candsBase.trackCandidateType()[trackCandidateIndex];
+      if (!(trackCandidateType == LSTObjType::T5 || trackCandidateType == LSTObjType::pT5))
+        return;
+
+      // Base quintuplet (for T5: objectIndices[0], for pT5: objectIndices[1])
+      const unsigned int refT5Index = (trackCandidateType == LSTObjType::T5)
+                                          ? candsExtended.objectIndices()[trackCandidateIndex][0]
+                                          : candsExtended.objectIndices()[trackCandidateIndex][1];
+
+      const float baseEta = __H2F(quintuplets.eta()[refT5Index]);
+      const float basePhi = __H2F(quintuplets.phi()[refT5Index]);
+      const uint8_t baseStartLogicalLayer = quintuplets.logicalLayers()[refT5Index][0];
+
+      // Module range to scan
+      int lowerModuleBegin = 0;
+      int lowerModuleEnd = static_cast<int>(modules.nLowerModules());
+
+      // If starting at layer 1, restrict to the module of the second hit
+      if (baseStartLogicalLayer == 1) {
+        lowerModuleBegin = quintuplets.lowerModuleIndices()[refT5Index][1];
+        lowerModuleEnd = lowerModuleBegin + 1;
+      }
+
+      const auto& threadIdx = alpaka::getIdx<alpaka::Block, alpaka::Threads>(acc);
+      const auto& blockDim = alpaka::getWorkDiv<alpaka::Block, alpaka::Threads>(acc);
+
+      // Flatten 2D thread indices within the block (Y, X) into one index
+      const int threadIndexFlat = threadIdx[1u] * blockDim[2u] + threadIdx[2u];
+      const int blockDimFlat = blockDim[1u] * blockDim[2u];
+
+      // Scan over lower modules
+      for (int lowerModuleIndex = lowerModuleBegin + threadIndexFlat; lowerModuleIndex < lowerModuleEnd;
```

Comment:
(following on an earlier comment)

> Cleaned this code up a bit to be similar to what Yanxi does in the T5 build kernel, but I'm not sure if there's something more clean we can replace this with.

I'm not sure that analogy applies
wouldn't a simple
`for (auto lowerModuleIndex : cms::alpakatools::uniform_elements(acc, lowerModuleEnd))` be enough ?

Reply:
I don't think this code example works because we start at lowerModuleBegin rather than 0, but I think this gets at the question of why I am using Acc3D when I flatten two of those dimensions. It looks like Acc1D works just as well, I'll push that soon.

Reply:
since the block direction is aligned with the candidates, perhaps a 2D and use a `uniform_elements_x` ?
the range can always be over `lowerModuleEnd-lowerModuleBegin` with an addition.

Note that I expect that having `lowerModuleBegin` not just 0 the saving is relatively minor and would go away when we start looking for overlap hits on the same layer